go-controller/pkg/node/gateway_udn_test.go

```diff
@@ -0,0 +1,175 @@
+package node
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/containernetworking/plugins/pkg/ns"
+	"github.com/containernetworking/plugins/pkg/testutils"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	ovntest "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/testing"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+)
+
+func getCreationFakeOVSCommands(fexec *ovntest.FakeExec, mgtPort, mgtPortMAC, netName, nodeName string, mtu int) {
+	fexec.AddFakeCmdsNoOutputNoError([]string{
+		"ovs-vsctl --timeout=15 -- --if-exists del-port br-int " + mgtPort +
+			" -- --may-exist add-port br-int " + mgtPort +
+			" -- set interface " + mgtPort +
+			" type=internal mtu_request=" + fmt.Sprintf("%d", mtu) +
+			" external-ids:iface-id=" + types.K8sPrefix + netName + "_" + nodeName,
+	})
+	fexec.AddFakeCmd(&ovntest.ExpectedCmd{
+		Cmd:    "ovs-vsctl --timeout=15 --if-exists get interface " + mgtPort + " mac_in_use",
+		Output: mgtPortMAC,
+	})
+	fexec.AddFakeCmdsNoOutputNoError([]string{
+		"ovs-vsctl --timeout=15 set interface " + mgtPort + " " + fmt.Sprintf("mac=%s", strings.ReplaceAll(mgtPortMAC, ":", "\\:")),
+	})
+}
+
+func getDeletionFakeOVSCommands(fexec *ovntest.FakeExec, mgtPort string) {
+	fexec.AddFakeCmdsNoOutputNoError([]string{
+		"ovs-vsctl --timeout=15 -- --if-exists del-port br-int " + mgtPort,
+	})
+}
+
+var _ = Describe("UserDefinedNetworkGateway", func() {
```

Comment:
I think we are missing testing the configured address with `LinkAddrAdd`

Reply:
oh good point! let me see if I can include that bit

Reply:
I tried and didn't succeed because we need netLink mocks... however good news is these tests already exist 
`func TestLinkAddrAdd(t *testing.T) {` and `func TestLinkAddrExist(t *testing.T) {` I will probably use the mocks here to help us

Reply:
ok done!